vscode-dotnet-runtime-library/src/Utils/WebRequestWorkerSingleton.ts

```diff
@@ -110,10 +108,68 @@ export class WebRequestWorker
 
         this.client = setupCache(uncachedAxiosClient,
             {
-                storage: buildMemoryStorage(),
-                ttl: this.cacheTimeToLive
+                storage: buildMemoryStorage(
+                    false,
+                ),
             }
         );
+
+        // This function must be a member of 'this' so it correctly 'binds' to the instance of the class and has access to class members in a different scope environment
+        this.cacheCleanupRunner = function ()
+        {
+            if (this?.client?.storage !== null && this?.client?.storage !== undefined)
+            {
+                try
+                {
+                    if ((process.hrtime.bigint() - this.lastCacheCleanTime) >= this.cacheTtl)
+                    {
+                        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+                        (this.client.storage as any)?.clear();
+                        this.lastCacheCleanTime = process.hrtime.bigint();
+                    }
+                }
+                catch (err)
+                {
+                    // clear returns a maybe promise which is a fake type where .catch does not work on it.
+                }
+            }
+            else
+            {
+                return true; // the class object was deleted externally (close tab / pkill from vscode ext service), so we should stop the cleanup.
+            }
+            return this.stopCacheCleanup.valueOf();
+        }
+
+        loopWithTimeoutOnCond(500, Number.POSITIVE_INFINITY,
+            this.cacheCleanupRunner,
+            function stopCleaning(): void {},
+            null,
+            new WebCacheClearEvent(`Clearing the web cache.`)
```

Comment:
this doesn't do anything but that's ok for now (event stream is null so its never posted)